### PR TITLE
Clean up hidden features notebook

### PR DESCRIPTION
**Pull request by cdraper-CS**
Use CatEncoder.  Stop using Normalizer, which normalizes per row. Remove coincidental explanations scan.

### CODE REVIEW

File: notebooks/hidden_features/HiddenFeatures.ipynb [hunk 1]
```diff
@@ -26,40 +26,22 @@
...
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "Accuracy of modal predictor: 0.7\n",
-      "Model 'aware' accuracy is 0.82\n",
-      "Model 'unaware' accuracy is 0.818\n"
+      "Model 'aware' accuracy is 0.781\n",
+      "Model 'unaware' accuracy is 0.867\n"
```
Comment (sdraper-CS) [inline notebooks/hidden_features/HiddenFeatures.ipynb:44]: This makes no sense.  No way should the unaware model have significantly higher accuracy!

**Review by sdraper-CS — COMMENTED** [review]
Certifai part seems ok, but the model accuracy seems VERY odd.

File: notebooks/hidden_features/HiddenFeatures.ipynb (hunk 1 above)
Comment (cdraper-CS) [reply notebooks/hidden_features/HiddenFeatures.ipynb]: Added test/train split, which makes everything better

Comment (akumar-cs) [reply notebooks/hidden_features/HiddenFeatures.ipynb]: typo "modal" -> model

File: notebooks/hidden_features/HiddenFeatures.ipynb [hunk 2]
```diff
@@ -26,40 +26,22 @@
...
     "                                      CertifaiPredictionTask, CertifaiTaskOutcomes, CertifaiOutcomeValue)\n",
-    "from certifai.scanner.report_utils import scores, construct_scores_dataframe"
+    "from certifai.scanner.report_utils import scores, construct_scores_dataframe\n",
+    "from certifai.common.utils.encoding import CatEncoder"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 2,
    "metadata": {},
```
Comment (akumar-cs) [inline notebooks/hidden_features/HiddenFeatures.ipynb:36]: typo in header "Firstly we'll build a model that does not have access to the protected features and see to **waht** degree" -> waht -> what

**Review by akumar-cs — APPROVED** [review]
looks good, apart from minor typos

File: notebooks/hidden_features/HiddenFeatures.ipynb [hunk 3]
```diff
@@ -40,8 +40,8 @@
      "output_type": "stream",
      "text": [
       "Accuracy of modal predictor: 0.7\n",
```
Comment (akumar-cs) [inline notebooks/hidden_features/HiddenFeatures.ipynb:42]: typo "modal" -> model

**Review by sdraper-CS — APPROVED** [review]
LGTM

**Pull request merged**